lib/diagnostic.js

```diff
@@ -47,7 +47,10 @@ let scopes = [/* your response here */];
 // executing it.
 //
 let fireOnAlderaan = function () {
-  // your code here
+  alderaan.fire = deathStar.fire.bind(alderaan); // haha
+  return alderaan.fire();
+
+  //return deathStar.fire.call(alderaan);
```

Comment:
Your second answer is exactly right; your first answer is close but sort of misses the mark.
If you're going to permanently attach a new `fire` method to `alderaan` (`alderaan.fire = `) then there's no need to do any binding to change the context.

Here's what you might write if you wanted to use `bind` instead of `call` or `apply`.
`return deathStar.fire.bind(alderaan)();`